roles/dnsmasq/tasks/manage_host.yml

```
@@ -14,72 +14,100 @@
 # License for the specific language governing permissions and limitations
 # under the License.
 
+- name: Ensure we have the right data and type
+  ansible.builtin.assert:
+    that:
+      - cifmw_dnsmasq_dhcp_entries is defined
+      - (cifmw_dnsmasq_dhcp_entries | type_debug) == 'list'
```

```suggestion
      - cifmw_dnsmasq_dhcp_entries is iterable
```

works better for this kind of tests See [here](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_tests.html#type-tests).

---

would love to know why they advise against using their own filters... comparing string might be "ugly", but `is iterable` alone wouldn't fit. To check if it's really a list, we'll have to use
`cifmw_dhcp_entries is not string and cifmw_dnsmasq_dhcp_entries is not mapping and cifmw_dnsmasq_dhcp_entries is iterable`

which is... well...